### PR TITLE
Add landing page for tutorials and reference guides

### PR DESCRIPTION
**Pull request by KaneFreeman**
## Description

- Add landing page (section landing) for tutorials and reference guides
- Modify manifest.json structure to include subsections, icons, descriptions and topics.
- Clean up styling for section lists, and current page highlighting

### Code

This PR touches:

- [ ] Content
- [x] Frontend
- [ ] Infrastructure
- [ ] Integration Tests

### Tests

- [x] Tests are included?

### Screenshots

image
image
image

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -8,6 +8,7 @@
 	<meta name="description" content="The home of dojo. Documentation. Examples. Community.">
 	<link rel="icon" href="./assets/favicon.ico" />
 	<link rel="stylesheet" type="text/css" href="./assets/prism.css" />
+	<link href="https://use.fontawesome.com/releases/v5.7.0/css/svg-with-js.css" rel="stylesheet"></link>
```
Comment (agubler) [inline src/index.html]: Does this work with BTR? I.e. I still don't think you'll be able to get the font awesome classes. Would it be better to import this into `main.css` so the fa classes can be inlined?

Comment (KaneFreeman) [reply src/index.html]: Yeah. That part is definitely still WIP. I am going to play around with it next OSS day and determine the best option for this.

**Comment by KaneFreeman** [comment]
@matt-gadd Using the long import paths for the icons drastically cut down on the main bundle size.

File: src/index.html (hunk 1 above)
Comment (KaneFreeman) [reply src/index.html]: I did end up removing this and going with the manual import as I was able to resolve the bundle size issue.

File: src/widgets/section/SectionLanding.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+const topicColors: { [key: string]: IconHeaderBackgroundColor } = {
+	dev: 'black',
+	cli: 'black',
+	themes: 'green',
+	forms: 'purple',
+	widgets: 'orange'
+};
+
+export default class SectionLanding extends WidgetBase<SectionLandingParameters> {
+	private _fetchSectionList() {
```
Comment (agubler) [inline src/widgets/section/SectionLanding.tsx]: Can we do this inline in the `render`?

File: src/widgets/section/SectionList.tsx [hunk 3]
```diff
@@ -2,41 +2,53 @@ import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
...
+	}
+
+	private _renderPageLink(page: PageDefinition) {
+		const { section, currentPath = '' } = this.properties;
+		const { path, name } = page;
+
+		const extraClasses: string[] = [css.itemLink];
+		const normalizedPath = path.replace(new RegExp(`^(.\/|..\/)*${section}\/`), '');
+		const normalizedCurrentPath = currentPath.replace(new RegExp(`^(.\/|..\/)*${section}\/`), '');
+		if (normalizedPath === normalizedCurrentPath) {
```
Comment (agubler) [inline src/widgets/section/SectionList.tsx]: Is there any reason why we can't use `ActiveLink` from `@dojo/framework/routing/ActiveLink` and pass the `css.selected` to the `activeClasses` property?

File: src/widgets/section/SectionLanding.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	private _renderPageCard(page: PageDefinition): DNode {
+		const { section } = this.properties;
+		const { name, path, description, icon, topic } = page;
+
+		return (
+			<div key={`${section}-page-${path}`} classes={css.pageLink}>
+				<LinkedCard
+					header={<CardIconHeader icon={icon} background={topicColors[topic]} />}
+					outlet={`${section}-page`}
+					params={{ page: path.replace(new RegExp(`^(.\/|..\/)*${section}\/`), '') }}
```
Comment (agubler) [inline src/widgets/section/SectionLanding.tsx]: How come we need these regular expressions throughout?

File: src/widgets/section/SectionLanding.tsx (hunk 2 above)
Comment (KaneFreeman) [reply src/widgets/section/SectionLanding.tsx]: Done

File: src/widgets/section/SectionLanding.tsx (hunk 4 above)
Comment (KaneFreeman) [reply src/widgets/section/SectionLanding.tsx]: Right now we are using the file path (will change this in the future) to determine the URL path. as such the section name is always there. I guess we could calculate this URL path in the section list block. Will adjust that.

**Comment by KaneFreeman** [comment]
@agubler Feedback has been addressed.

**Pull request merged**